src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -65,5 +67,34 @@ public IActionResult GetInstanceInfo()
         {
             return Ok(_instanceManager.GetInstanceInfo());
         }
+
+        /// <summary>
+        /// This endpoint generates a temporary x-ms-site-restricted-token for core tool
+        /// to access KuduLite zipdeploy endpoint in Linux Consumption
+        /// </summary>
+        /// <returns>
+        /// 200 on token generated
+        /// 400 on non-Linux container environment
+        /// 404 on WEBSITE_AUTH_ENCRYPTION_KEY is not set
+        /// </returns>
+        [HttpGet]
+        [Route("admin/getsitetoken")]
```

Comment:
I think you should move this to HostController under route admin/host/token (similar to naming used in Kudu route [here](https://github.com/projectkudu/kudu/blob/a421963dc24a9dcef0f0ca84bac5739f3bf132b8/Kudu.Services.Web/App_Start/NinjectServices.cs#L569)).